test/EFCore.PG.FunctionalTests/TestUtilities/TestNpgsqlConnection.cs

```diff
@@ -52,7 +52,7 @@ void PreOpen()
 
             if (fail.HasValue)
             {
-                throw new NpgsqlException();
+                throw new PostgresException("Simulated failure", "ERROR", "ERROR", ErrorCode);
```

Comment:
By the way, I thought to make its ctor private.

Reply:
That is problematic for testing :) In the past there were also requests from users to be able to test their code by creating PostgresException themselves.

Reply:
Reflection is their friend (:

Reply:
Is there any advantage in making the ctor private?